cuda/private/rules/cuda_library.bzl

```diff
@@ -81,6 +81,18 @@ def _cuda_library_impl(ctx):
     libs = [] if lib == None else [lib]
     pic_libs = [] if pic_lib == None else [pic_lib]
 
+    cc_infos = [
+        CcInfo(
+            compilation_context = compilation_ctx,
+            linking_context = linking_ctx,
+        ),
+    ]
+
+    if hasattr(ctx.attr, "_builtin_deps"):
+        cc_infos.extend([dep[CcInfo] for dep in ctx.attr._builtin_deps if CcInfo in dep])
+
+    cc_info = cc_common.merge_cc_infos(cc_infos = cc_infos, direct_cc_infos = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep])
```

Comment:
Assume you return `merged_cc_info` from the `cuda_common.merged_cc_info`, I think this should be written/shorten as
```starlark
cc_info = cc_common.merge_cc_infos(cc_infos = common.merged_cc_info, direct_cc_infos = [CcInfo(
            compilation_context = compilation_ctx,
            linking_context = linking_ctx,
        )])
```

Reply:
Maybe call it as `cuda_common.merged_transitive_cc_info` to make it clear that it is not direct :)

Reply:
I went with `transitive_cc_info` to mirror `transitive_linking_contexts` which already existed.